mlflow/pipelines/steps/predict.py

```diff
@@ -59,11 +59,21 @@ def _run(self, output_directory):
 
         # score dataset
         model_uri = self.step_config["model_uri"]
-        predict = mlflow.pyfunc.spark_udf(spark, model_uri)
+        env_manager = "local" if "_disable_env_restoration" in self.step_config else "conda"
+        predict = mlflow.pyfunc.spark_udf(spark, model_uri, env_manager=env_manager)
         scored_sdf = input_sdf.withColumn("prediction", predict(struct(*input_sdf.columns)))
 
         # save predictions
-        # TODO: add Delta and Table output formats
+        # note: the current output writing logic allows no overwrites
+        output_format = self.step_config["output_format"]
+        if output_format == "parquet" or output_format == "delta":
+            scored_sdf.coalesce(1).write.format(output_format).save(
```

Comment:
Do we check whether we are overwriting existing data?

Reply:
This will fail if we are writing to a location where data already exists.

Reply:
We could add an overwrite option in the future, but for now, overwrite is not allowed.